### PR TITLE
dioxus-cli: 0.5.7 -> 0.6.0

### PR DESCRIPTION
**Pull request by CathalMullan**
## Description of changes

- updated `dioxus-cli` to v0.6.0
- enabled `optimizations` cargo feature
- formatted with `nixfmt-rfc-style`
- added changelog

Tested by running a [Desktop, Web, Android and iOS](https://i.imgur.com/d2KTWFi.png) version of the hello world example simultaneously on aarch64-darwin. 

[Here's the flake I used](https://gist.github.com/CathalMullan/95f41f38e485ee90947c67167cf3bc7a).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/di/dioxus-cli/package.nix [hunk 1]
```diff
@@ -1,42 +1,49 @@
...
 
   src = fetchCrate {
     inherit pname version;
-    hash = "sha256-/LeMh5WX4dvkveu5w6qBQLbtoi5yUW6iad0YatA/tMQ=";
+    hash = "sha256-0Kg2/+S8EuMYZQaK4Ao+mbS7K48VhVWjPL+LnoVJMSw=";
   };
 
-  cargoHash = "sha256-D6y2NiFqSf0u6icSKCRZK7ycR+GswOX627M7PEy/D6U=";
+  cargoHash = "sha256-RMo6q/GSAV1bCMWtR+wu9xGKCgz/Ie6t/8oirBly/LQ=";
+  cargoBuildFeatures = [ "optimizations" ];
```
Comment (iynaix) [inline pkgs/by-name/di/dioxus-cli/package.nix:26]: This should be `buildFeatures`.

**Review by iynaix — APPROVED** [review]

**Review by xanderio — APPROVED** [review]
The ofborg failure seams unrelated, I'm going to check it when i'm at my Laptop if so i'm going to merge this.

**Comment by xanderio** [comment]
> The ofborg failure seams unrelated, I'm going to check it when i'm at my Laptop if so i'm going to merge this.

This was fixed in #364542.

**Comment by xanderio** [comment]
@CathalMullan I'm not sure if there are breaking changes in the CLI interface or just that the interactive parts of the CLI look different, if there aren't any breaking changes we can backport this to 24.11.

**Comment by CathalMullan** [comment]
@xanderio The overall CLI interface hasn't changed much, but there are some breaking changes around options that I can see. e.g. `--platform fullstack` in v0.5.7 has been changed to `--fullstack`. So I wouldn't backport.

**Comment by cafkafk** [comment]
`dx new` seems to just fail now... and seems ofborg failed?

**Comment by CathalMullan** [comment]
> `dx new` seems to just fail now

Seems to works fine for me on aarch64-linux?
What command are you running?

```
[cmullan@home] ~/workspace/nixpkgs master
> nix run .#dioxus-cli -- --version                                                                                                                13:28:36
dioxus 0.6.0 (was built without git repository)
[cmullan@home] ~/workspace/nixpkgs master
> nix run .#dioxus-cli -- new project                                                                                                              13:28:41
✔ 🤷   Which sub-template should be expanded? · Workspace
✔ 🤷   Do you want to use Dioxus Fullstack? · true
✔ 🤷   Do you want to use Dioxus Router? · true
[cmullan@asahi] ~/workspace/nixpkgs master
> tree project/                                                                                                                          (took 4s) 13:28:50
project/
├── Cargo.toml
├── README.md
├── desktop
│   ├── Cargo.toml
│   ├── assets
│   │   ├── blog.css
│   │   └── main.css
│   └── src
│       ├── main.rs
│       └── views
│           ├── blog.rs
│           ├── home.rs
│           └── mod.rs
├── mobile
│   ├── Cargo.toml
│   ├── assets
│   │   ├── blog.css
│   │   └── main.css
│   └── src
│       ├── main.rs
│       └── views
│           ├── blog.rs
│           ├── home.rs
│           └── mod.rs
├── server
│   ├── Cargo.toml
│   └── src
│       └── lib.rs
├── ui
│   ├── Cargo.toml
│   ├── assets
│   │   ├── header.svg
│   │   └── styling
│   │       ├── echo.css
│   │       ├── hero.css
│   │       └── navbar.css
│   └── src
│       ├── echo.rs
│       ├── hero.rs
│       ├── lib.rs
│       └── navbar.rs
└── web
    ├── Cargo.toml
    ├── assets
    │   ├── blog.css
    │   ├── favicon.ico
    │   └── main.css
    └── src
        ├── main.rs
        └── views
            ├── blog.rs
            ├── home.rs
            └── mod.rs

18 directories, 35 files
```

**Comment by cafkafk** [comment]
```
dx new project
   1.551s ERROR err=Other(Please check if the Git user / repository exists.

Caused by:
    could not read from remote repository; class=Net (12); code=Eof (-20)) 
```

which is likely this issue https://github.com/DioxusLabs/dioxus/issues/3319

**Comment by CathalMullan** [comment]
Strangely seems to work fine for me on aarch64-darwin too. 

```
> nix build .#dioxus-cli                                                                                                     (took 4s) 13:45:36
[cmullan@mbp] ~/workspace/nixpkgs master
> nix run .#dioxus-cli -- new project                                                                                        (took 4s) 13:45:36
✔ 🤷   Which sub-template should be expanded? · Workspace
✔ 🤷   Do you want to use Dioxus Fullstack? · true
✔ 🤷   Do you want to use Dioxus Router? · true
```

**Comment by CathalMullan** [comment]
Ah, if I change my git config to:

```
[url "git@github.com:"]
insteadOf = https://github.com/
```

I can re-produce the error.

**Pull request merged**